internal/components/setup/kind.go

```diff
@@ -73,8 +77,76 @@ type kindPort struct {
 	waitExpose string // Need to use when expose
 }
 
-//nolint:gocyclo // skip the cyclomatic complexity check here
+func listLocalImages(ctx context.Context, cli *docker.Client) (map[string]struct{}, error) {
+	summary, err := cli.ImageList(ctx, types.ImageListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	res := make(map[string]struct{}, len(summary))
+	for i := 0; i < len(summary); i++ {
+		tags := summary[i].RepoTags
+		for j := 0; j < len(tags); j++ {
+			res[tags[j]] = struct{}{}
+		}
+	}
+	return res, nil
+}
+
+// pullImages pull docker image from a private repository
+func pullImages(ctx context.Context, images []string) error {
+	cli, err := docker.NewClientWithOpts(docker.FromEnv)
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	localImages, err := listLocalImages(ctx, cli)
+	if err != nil {
+		logger.Log.Error("list local images error", "error", err)
```

Comment:
Since the error is returned and logged in the caller, we don't need to log here. A simple `errors.Wrap` can be used.